### DISCUSSION TITLE
CVE-2023-38545...

### DISCUSSION THREAD

**Discussion by whrds** [community/community#73261] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hello,
I'm currently trying to analyze 'CVE-2023-38545', but when I try to practice with the POC or open a proxy server to practice, I don't get a segment fault error. According to the CVE details, it should occur when a host name of more than 256 bytes is entered, but when I input it, I get:

curl: (97) Could not resolve host: aaaaaaaaaaaaaaaaa....
![image](https://github.com/community/community/assets/35793143/5275deda-beda-4463-8e16-8f0b41cf7d7c)

If anyone knows about this CVE, I'm curious about how and in what environment you practiced.

**Comment by 0x48piraj** [comment, marked as answer]
This vulnerability affects curl/libcurl only in limited cases. This is because the vulnerability only exists when curl/libcurl is used through a SOCKS5 proxy with a specific set of configurations.

An overflow is only possible in applications that do not set CURLOPT_BUFFERSIZE or set it smaller than 65541. Since the curl tool sets CURLOPT_BUFFERSIZE to 100kB by default it is not vulnerable unless rate limiting was set by the user to a rate smaller than 65541 bytes/second.

The options that cause SOCKS5 with remote hostname to be used in libcurl:

CURLOPT_PROXYTYPE set to type CURLPROXY_SOCKS5_HOSTNAME, or:
CURLOPT_PROXY or CURLOPT_PRE_PROXY set to use the scheme socks5h://
One of the proxy environment variables can be set to use the socks5h:// scheme. For example http_proxy, HTTPS_PROXY or ALL_PROXY.

The options that cause SOCKS5 with remote hostname to be used in the curl tool:

--socks5-hostname, or:
--proxy or --preproxy set to use the scheme socks5h://
Environment variables as described in the libcurl section.


Seems like this is a good repository on a cursory reading: https://github.com/UTsweetyfish/CVE-2023-38545
